spec/deaf_grandma_spec.rb

```diff
@@ -9,10 +9,19 @@
   end
   
   it "says 'NOT SINCE 1964!' when we yell" do
-    #implement your test here
+    expect(script.speak("HELLO!")).to eq "NOT SINCE 1964!"
   end
 
-  it "EXTRA CREDIT: How would you test yelling BYE?" do
-    #implement your test here
+  it "says 'SEE YOU LATER SONNY! when we say 3 consecutive 'BYE's" do
+    expect(script.speak("BYE")).to eq "NOT SINCE 1964!"
+    expect(script.speak("BYE")).to eq "NOT SINCE 1964!"
+    expect(script.speak("BYE")).to eq "SEE YOU LATER SONNY!"
+  end
+
+  it "says 'NOT SINCE 1964! when we say 3 'BYE's nonconsecutively" do
+    expect(script.speak("BYE")).to eq "NOT SINCE 1964!"
+    expect(script.speak("BYE")).to eq "NOT SINCE 1964!"
+    expect(script.speak("I guess I'll stay")).to eq "SPEAK UP SONNY!"
+    expect(script.speak("BYE")).to eq "NOT SINCE 1964!"
```

Comment:
There are interesting semi-theoretical questions here about what should be under test in these two examples.  If it's really what the third response is, given the setup, then why check the expectation on the first two `speak("BYE")`'s?  If it's all three, then maybe too much is being covered in a single test.

Reply:
I agree. I haven't done much work on testing so I'm still working my way through it. It is aiming to test that saying 'BYE' 3 times non consecutively doesn't change the output. I'll update now

Reply:
🏁